diff --git a/lib/graphics.js b/lib/graphics.js
index f7438a0b93e31a528c943bf2aa0f61a8e4d99af4..f11dfe5e00b776bd5e6921c1d7a937d0e53d6ee0 100644
--- a/lib/graphics.js
+++ b/lib/graphics.js
@@ -14,6 +14,7 @@
 // ----------------------------------------------------------------------------------
 
 const fs = require('fs');
+const path = require('path');
 const exec = require('child_process').exec;
 const execSync = require('child_process').execSync;
 const util = require('./util');
@@ -422,25 +423,24 @@ function graphics(callback) {
 
     if (_windows) {
       try {
-        const basePath = util.WINDIR + String.raw`\System32\DriverStore\FileRepository`;
-        // find all directories that have an nvidia-smi.exe file
-
-        const candidateDirs = fs.readdirSync(basePath).filter((dir) => {
-          if (fs.statSync([basePath, dir].join('/')).isDirectory()) {
-            return fs.readdirSync([basePath, dir].join('/')).includes('nvidia-smi.exe');
-          } else {
-            return false;
-          }
-        });
-        // use the directory with the most recently created nvidia-smi.exe file
-        const targetDir = candidateDirs.reduce((prevDir, currentDir) => {
-          const previousNvidiaSmi = fs.statSync([basePath, prevDir, 'nvidia-smi.exe'].join('/'));
-          const currentNvidiaSmi = fs.statSync([basePath, currentDir, 'nvidia-smi.exe'].join('/'));
-          return previousNvidiaSmi.ctimeMs > currentNvidiaSmi.ctimeMs ? prevDir : currentDir;
-        });
-
-        if (targetDir) {
-          _nvidiaSmiPath = [basePath, targetDir, 'nvidia-smi.exe'].join('/');
+        const basePath = path.join(util.WINDIR, 'System32', 'DriverStore', 'FileRepository');
+        // find all directories that have an nvidia-smi.exe file with date
+        const candidates = fs
+          .readdirSync(basePath, { withFileTypes: true })
+          .filter((dir) => dir.isDirectory())
+          .map((dir) => {
+            const nvidiaSmiPath = path.join(basePath, dir.name, 'nvidia-smi.exe');
+            try {
+              const stats = fs.statSync(nvidiaSmiPath);
+              return { path: nvidiaSmiPath, ctime: stats.ctimeMs };
+            } catch {
+              return null;
+            }
+          })
+          .filter(Boolean);
+        if (candidates.length > 0) {
+          // take the most recent
+          _nvidiaSmiPath = candidates.reduce((prev, curr) => (curr.ctime > prev.ctime ? curr : prev)).path;
         }
       } catch {
         util.noop();
@@ -457,12 +457,12 @@ function graphics(callback) {
     if (nvidiaSmiExe) {
       const nvidiaSmiOpts =
         '--query-gpu=driver_version,pci.sub_device_id,name,pci.bus_id,fan.speed,memory.total,memory.used,memory.free,utilization.gpu,utilization.memory,temperature.gpu,temperature.memory,power.draw,power.limit,clocks.gr,clocks.mem --format=csv,noheader,nounits';
-      const cmd = nvidiaSmiExe + ' ' + nvidiaSmiOpts;
+      const cmd = `"${nvidiaSmiExe}"` + ' ' + nvidiaSmiOpts;
       if (_linux) {
         options.stdio = ['pipe', 'pipe', 'ignore'];
       }
       try {
-        const sanitized = util.sanitizeShellString(cmd) + (_linux ? '  2>/dev/null' : '') + (_windows ? '  2> nul' : '');
+        const sanitized = cmd + (_linux ? '  2>/dev/null' : '') + (_windows ? '  2> nul' : '');
         const res = execSync(sanitized, options).toString();
         return res;
       } catch {
