diff --git a/lib/graphics.js b/lib/graphics.js
index f7438a0b93e31a528c943bf2aa0f61a8e4d99af4..0a734c04a6d2328e6c2ad3a9060915d5a48e817d 100644
--- a/lib/graphics.js
+++ b/lib/graphics.js
@@ -422,25 +422,24 @@ function graphics(callback) {
 
     if (_windows) {
       try {
-        const basePath = util.WINDIR + String.raw`\System32\DriverStore\FileRepository`;
-        // find all directories that have an nvidia-smi.exe file
-
-        const candidateDirs = fs.readdirSync(basePath).filter((dir) => {
-          if (fs.statSync([basePath, dir].join('/')).isDirectory()) {
-            return fs.readdirSync([basePath, dir].join('/')).includes('nvidia-smi.exe');
-          } else {
-            return false;
-          }
-        });
-        // use the directory with the most recently created nvidia-smi.exe file
-        const targetDir = candidateDirs.reduce((prevDir, currentDir) => {
-          const previousNvidiaSmi = fs.statSync([basePath, prevDir, 'nvidia-smi.exe'].join('/'));
-          const currentNvidiaSmi = fs.statSync([basePath, currentDir, 'nvidia-smi.exe'].join('/'));
-          return previousNvidiaSmi.ctimeMs > currentNvidiaSmi.ctimeMs ? prevDir : currentDir;
-        });
-
-        if (targetDir) {
-          _nvidiaSmiPath = [basePath, targetDir, 'nvidia-smi.exe'].join('/');
+        const basePath = path.join(util.WINDIR, 'System32', 'DriverStore', 'FileRepository');
+        // find all directories that have an nvidia-smi.exe file with date
+        const candidates = fs
+          .readdirSync(basePath, { withFileTypes: true })
+          .filter((dir) => dir.isDirectory())
+          .map((dir) => {
+            const nvidiaSmiPath = path.join(basePath, dir.name, 'nvidia-smi.exe');
+            try {
+              const stats = fs.statSync(nvidiaSmiPath);
+              return { path: nvidiaSmiPath, ctime: stats.ctimeMs };
+            } catch {
+              return null;
+            }
+          })
+          .filter(Boolean);
+        if (candidates.length > 0) {
+          // take the most recent
+          _nvidiaSmiPath = candidates.reduce((prev, curr) => (curr.ctime > prev.ctime ? curr : prev)).path;
         }
       } catch {
         util.noop();
