diff --git a/node_modules/systeminformation/lib/graphics.js b/node_modules/systeminformation/lib/graphics.js
index 0cc7a0f..6bb6ed3 100644
--- a/node_modules/systeminformation/lib/graphics.js
+++ b/node_modules/systeminformation/lib/graphics.js
@@ -394,7 +394,9 @@ function graphics(callback) {
         const basePath = util.WINDIR + '\\System32\\DriverStore\\FileRepository';
         // find all directories that have an nvidia-smi.exe file
         const candidateDirs = fs.readdirSync(basePath).filter(dir => {
-          return fs.readdirSync([basePath, dir].join('/')).includes('nvidia-smi.exe');
+          if (fs.statSync([basePath, dir].join('/')).isDirectory()) {
+            return fs.readdirSync([basePath, dir].join('/')).includes('nvidia-smi.exe');
+          }
         });
         // use the directory with the most recently created nvidia-smi.exe file
         const targetDir = candidateDirs.reduce((prevDir, currentDir) => {
